templates/CONTRIBUTING.md

```diff
@@ -0,0 +1,115 @@
+* **For bugs**, if you have already found a fix, feel free to submit a Pull Request referencing the Issue you created. Include the `Fixes #` syntax to link it to the issue you're addressing.
+* **For feature requests**, we want to improve upon the library incrementally which means small changes at a time. In order to ensure your PR can be reviewed in a timely manner, please keep PRs small. If you think this is unrealistic, then mention that within the issue and we can discuss it.
+
+## Workflow
+
+Once you're ready to contribute code back to this repo, ensure you setup your environment to be prepared for upstream contributions.
+
+### 1) Fork on GitHub
+
+Fork the appropriate repository by clicking the Fork button (top right) on GitHub.
+
+### 2) Create a Local Fork
+
+From whatever directory you want to have this code, clone this repository and setup some sane defaults:
+
+```
+$ git clone https://github.com/<you-username>/<this repo>
+# or: git clone git@github.com:$user/<this repo>.git
+
+$ cd <this repo>
+
+$ git remote add upstream https://github.com/sensu-plugins/<this repo>.git
+# or: git remote add upstream git@github.com:sensu-plugins/<this repo>.git
+
+# Never allow a push to upstream master
+$ git remote set-url --push upstream no_push
+
+# Confirm that your remotes make sense:
+$ git remote -v
+```
+
+### 3) Create a Branch for Your Contribution
+
+Begin by updating your local fork of the plugin:
+
+```
+$ git fetch upstream
+$ git checkout master
+$ git rebase upstream/master
+```
+
+Create a new, descriptively named branch to contain your change:
+
+```
+$ git checkout -b feature/myfeature
+```
+
+Now hack away at your awesome feature on the `feature/myfeature` branch.
+
+### 4) Testing Your Code
+
+There is not yet a standardization on how we test plugins across Sensu Community contributions. If you'd like to be part of the solution, [comment on this open Issue](https://github.com/sensu-plugins/community/issues/46). For now, please run your code on Sensu and ensure it works as expected.
+
+### 5) Committing Code
+
+Commit your changes with a thoughtful commit message.
+
+```
+$ git commit -am "Adding a check for Foo
+
+Foo is a particularly helpful status when working with Bar. Designed to gather XYZ from the foobar interface."
+```
+
+Repeat the commit process as often as you need and then edit/test/repeat. Minor edits can be added to your last commit quite easily:
+
+```
+$ git add -u
+$ git commit --amend
+```
+
+### 6) Pushing to GitHub
+
+When ready to review (or just to establish an offsite backup or your work), push your branch to your fork on GitHub:
+
+```
+$ git push origin feature/myfeature
+```
+
+If you recently used `commit --amend`, you may need to force push:
+
+```
+$ git push -f origin feature/myfeature
+```
+
+### 7) Create a Pull Request
+
+Create a pull request by visiting https://github.com/<your-username>/<this-repo> and following the instructions at the top of the screen.
```

Comment:
when previewing the markdown it looks like `https://github.com//` perhaps we should wrap it with a pair of backticks or choose a symbol such as `$` rather than `<>`.

Reply:
Great point. I actually replaced it with the token substitution format used for https://github.com/mbbroberg/snap-pluginsync which I intend to use to distribute these to every plugin. I'll test it before running it, I promise 😆